src/Core/Validation/FieldValidation/FieldValidatorsTrait.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use RuntimeException;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Validation\FieldValidation\FieldValidationInterface;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\FormField;
+
+trait FieldValidatorsTrait
+{
+    /**
+     * FieldValidators configuration for the field, which is either a FormField or DBField
+     *
+     * Each item in the array can be one of the following
+     * a) MyFieldValidator::class,
+     * b) MyFieldValidator::class => [null, 'getMyArg'],
+     * c) MyFieldValidator::class => null,
+     *
+     * a) Will create a FieldValidator and pass the name and value of the field as args to the constructor
+     * b) Will create a FieldValidator and pass the name, value, make a pass additional args, calling each
+     *    non-null value on the field e.g. it will skip the first arg and call $field->getMyArg() for the second arg
+     * c) Will disable a previously set FieldValidator. This is useful to disable a FieldValidator that was set
+     *    on a parent class
+     *
+     * You may only have a single instance of a FieldValidator class per field
+     */
+    private static array $field_validators = [];
+
+    /**
+     * Validate this field
+     */
+    public function validate(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        // Skip validation if the field value is null
+        if ($this->getValue() === null) {
+            return $result;
+        }
+        $fieldValidators = $this->getFieldValidators();
+        foreach ($fieldValidators as $fieldValidator) {
+            $validationResult = $fieldValidator->validate();
+            if (!$validationResult->isValid()) {
+                $result->combineAnd($validationResult);
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Get FieldValidators based on `field_validators` configuration
+     */
+    private function getFieldValidators(): array
+    {
+        $fieldValidators = [];
+        // Used to disable a validator that was previously set with an int index
+        $disabledClasses = [];
+        $interface = FieldValidationInterface::class;
+        // temporary check, will make FormField implement FieldValidationInterface in a future PR
+        $tmp = FormField::class;
+        if (!is_a($this, $interface) && !is_a($this, $tmp)) {
```

Comment:
```suggestion
        // temporary check, will make FormField implement FieldValidationInterface in a future PR
        $tmp = FormField::class;
        if (!is_a($this, FieldValidationInterface::class) && !is_a($this, $tmp)) {
```
Seems like this is the only time that variable is used, so just putting the reference in place makes it _much_ easier to see what's happening.

Edit: Okay I see why the var was used but I still don't think it should be. See https://github.com/silverstripe/silverstripe-framework/pull/11408#discussion_r1811560562

Reply:
Removed